app/services/hyrax/versioning_service.rb

```diff
@@ -53,6 +49,16 @@ def latest_version
       versions.last
     end
 
+    ##
+    # Returns whether support for multiple versions exists on this
+    # +Hyrax::VersioningService+.
+    #
+    # Versioning is unsupported on nil resources or on Valkyrie resources when
+    # the configured storage adapter does not advertise versioning support.
+    def supports_multiple_versions?
+      !(resource.nil? || resource.is_a?(Hyrax::FileMetadata) && !storage_adapter.try(:"supports?", :versions))
```

Comment:
is the assumption that if it's not a `FileMetadata` it's an AF thing?

Reply:
This is the existing assumption in the codebase, yeah. This is just copying the behaviour of the previous `VersioningService` implementation (which I wrote), but my recollection is that I was following patterns I saw elsewhere.

Reply:
seems good. just making sure we weren't trying to support other valkyrie models